### PR TITLE
Preferences cleanup

### PR DESCRIPTION
**Pull request by pbn4**
# Pull Request Template

## Issue

Addresses # (#1837 )

- [ ] This change addresses the issue in full
- [x] This change addresses only certain aspects of the issue
- [ ] This change is a dependency for another issue
- [ ] This change has a dependency from another issue

## Description

Introduce N-M Listing-Preference relation through a self managed (not TypeORM managed) intermediate entity ListingPreference, which now holds ordinal and page. Remove Preference entity entirely with an appropriate DB migration. 

@emilyjablonski @dominikx96 @jaredcwhite This is a breaking change for everything related with retrieving preferences through Listing model. 

Listing.preference data has been moved to Listing.listingPreferences[].preference. `Page` and `Ordinal` are moved from Preference to ListingPreference. Example:

```
{
  ...listing,
  listingPreferences: [
    {
      preference: {id: preferenceId, title: ...}
      ordinal: 1,
      page: 1
    },
    {
      preference: {id: preferenceId2, title: ...}
      ordinal: 2,
      page: 2
    }
  ]
}
```

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Prototype/POC (not to merge)
- [x] This change is a refactor/address technical debt
- [ ] This change requires a documentation update
- [ ] This change requires a SQL Script

## How Can This Be Tested/Reviewed?

Places preferences are used:
* Displayed on the individual listing view (public)
* Submitting an application (public)
* Viewing an application (partners)
* Editing an application (partners)
* Adding and editing preferences on a listing (partners)

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have reviewed the changes in a desktop view
- [ ] I have reviewed the changes in a mobile view
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [x] I have assigned reviewers
- [x] I have updated the changelog to include a description of my changes
- [x] I have run `yarn generate:client` if I made backend changes

### CODE REVIEW

File: backend/core/src/preferences/preferences.service.ts [hunk 1]
```diff
@@ -1,9 +1,62 @@
...
+import { assignDefined } from "../shared/assign-defined"
+import { addFilters } from "../shared/filter"
+import { PreferencesListQueryParams } from "./dto/preferences-list-query-params"
+import { PreferencesFilterParams } from "./dto/preferences-filter-params"
+import { jurisdictionFilterTypeToFieldMap } from "./dto/jurisdictionFilterTypeToFieldMap"
+
+export class PreferencesService {
+  constructor(@InjectRepository(Preference) private readonly repository: Repository<Preference>) {}
+
+  list(params?: PreferencesListQueryParams): Promise<Preference[]> {
```
Comment (emilyjablonski) [inline backend/core/src/preferences/preferences.service.ts:17]: How do I fetch preferences if a user has more than one jurisdiction?

**Review by seanmalbert — CHANGES_REQUESTED** [review]
@pbn4 and @emilyjablonski ,
There are two pieces that seem off:

1. In partners, if I remove preferences and add them to a listing `listingPreference.page` becomes null, which means they won't show up on the application. We don't have a UI for setting the page of a selected preference, and until we do, we may need to keep a page column on `preferences` too, so we can set that on `listings_preferences` when adding to a listing. We should bring this up in stand-up, but it may be the best solution until we have an improved UI for adding preferences on a listing
2. This next one isn't a bug, but I want to call it out here along with item 1. When adding a preference to a listing and once it's added, there isn't any indication in partners that the preference will be hidden on the listing detail. Even once we have a UI for adding preferences, it would be nice to have an indicator on the LM form/detail that the preference will be hidden.

**Comment by seanmalbert** [comment]
As discussed, we're going to keep pages for now and we need to update the LM form, so that the listingPreferences getting set are using the page value from the preference, so that preferences show on the application. In a separate issue, we are going to remove pages and update the application to split preferences across pages based off their ordinal.

**Comment by emilyjablonski** [comment]
@pbn4 had some significant backend merge conflicts w the new jurisdiction language piece but i think i resolved them ok

**Comment by seanmalbert** [comment]
removed the no longer needed test and updated others, since preferences are no longer on the same page.

**Pull request merged**